www/pfSense-pkg-vHosts/files/usr/local/www/vhosts.php

```diff
@@ -0,0 +1,422 @@
+
+	/*-----------------------------*/
+	/* Set the form action if any. */
+	/*-----------------------------*/
+	if (isset($form_action)) {
+		$form->setAction($form_action);
+	}
+
+	foreach($sections as $s) {
+		foreach($s as $title => $fields) {
+			$section = new Form_Section($title);
+
+			foreach($fields as $field) {
+				$section->addInput($field->setHelp($help[$field->getName()] ?: null));
+			}
+			$form->add($section);
+		}
+	}
+
+	/*----------------------------------*/
+	/* Cancel button reloads list page. */
+	/*----------------------------------*/
+	$cancel = new Form_Button('cancel', 'Cancel', null, 'fa-undo');
+	$cancel->setAttribute('type', 'button');
+	$cancel->setAttribute('onclick', 'window.location = "?"');
+	$cancel->addClass('btn-warning');
+	$form->addGlobal($cancel);
+
+	/*-----------------*/
+	/* Drop form page. */
+	/*-----------------*/
+	$helpurl = "https://doc.pfsense.org/index.php/vHosts_package?";
+	include("head.inc");
+
+	if ($input_errors) {
+		print_input_errors($input_errors);
+	}
+	
+	display_top_tabs($pgtabs);
+	
+	print $form;
+}
+
+/*******************************************************************************
+* List Page
+*******************************************************************************/
+else {
+	$shortcut_section = $vhosts_g['subsystem_name'];
+	$vhostsd_running  = vhostsd_is_running();
+
+	include("head.inc");
+
+	if (vhosts_dirty()) {
+		print_info_box(gettext('vHosts configuration has been changed. Restart service to apply changes.'));
+	}
+
+	display_top_tabs($pgtabs);
+
+	?>
+	<div class="panel panel-default">
+		<div class="panel-heading"></div>
+		<div class="panel-body table-responsive">
+			<table class="table table-striped table-hover table-condensed sortable-theme-bootstrap table-rowdblclickedit" data-sortable>
+				<thead>
+					<tr>
+						<th><?=gettext("IP Address:Port")?></th>
+						<th><?=gettext("Directory Name")?></th>
+						<th><?=gettext("Host Name")?></th>
+						<th><?=gettext("Secure")?></th>
+						<th><?=gettext("Description")?></th>
+					</tr>
+				</thead>
+				<tbody>
+				<?php
+
+				for ($i=0; $i < count($a_vhosts); $i++) {
+					$vhost     = $a_vhosts[$i];
+					$disabled  = isset($vhost['disabled']);
+					$dirname   = $vhost['dirname'];
+					$hostname  = $vhost['hostname'];
+					$ipaddress = $vhost['ipaddress'];
+					$port      = $vhost['port'];
+					$ssl       = vhosts_lookup_cert($vhost['certref']);
+
+					/*-------------------------------------------------*/
+					/* Build link to vhost around host name. Use first */
+					/* host listed in 'hostname' or 'ipaddress:port'.  */
+					/*-------------------------------------------------*/
+					$ipport    = htmlspecialchars("$ipaddress:$port");
+					$hostname  = ($hostname ? explode(' ', $hostname)[0] : '');
+					$hosturl   = ($ssl ? 'https' : 'http')."://".($hostname ?: $ipaddress).":$port";
+					$hostlink  = ($disabled || !$vhostsd_running ? $ipport : "<a href='$hosturl'>$ipport</a>");
```

Comment:
You encoded `$ipport` but you also need to encode `$hosturl` before it is used in this way, too.